config/load.go

```diff
@@ -0,0 +1,210 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/joho/godotenv"
+	"github.com/spf13/viper"
+)
+
+func (c *Config) load() {
+	c.hotReloadableConfig = make(map[string][]*configValue)
+	c.nonHotReloadableConfig = make(map[string][]*configValue)
+	c.envs = make(map[string]string)
+
+	if err := godotenv.Load(); err != nil {
+		fmt.Println("INFO: No .env file found.")
+	}
+
+	configPath := getEnv("CONFIG_PATH", "./config/config.yaml")
+
+	v := viper.NewWithOptions(viper.EnvKeyReplacer(&envReplacer{c: c}))
+	v.AutomaticEnv()
+	bindLegacyEnv(v)
+
+	v.SetConfigFile(configPath)
+	err := v.ReadInConfig() // Find and read the config file
+	// Don't panic if config.yaml is not found or error with parsing. Use the default config values instead
+	if err != nil {
+		fmt.Printf("[Config] :: Failed to parse config file from path %q, using default values: %v\n", configPath, err)
+	}
+	v.OnConfigChange(func(e fsnotify.Event) {
+		c.onConfigChange()
+	})
+	v.WatchConfig()
+
+	c.v = v
+}
+
+func (c *Config) onConfigChange() {
+	defer func() {
+		if r := recover(); r != nil {
+			err := fmt.Errorf("cannot update Config Variables: %v", r)
+			fmt.Println(err)
+		}
+	}()
+	c.vLock.RLock()
+	defer c.vLock.RUnlock()
+	c.mapLock.RLock()
+	defer c.mapLock.RUnlock()
+	_ = c.checkAndHotReloadConfig(c.hotReloadableConfig)
+	isChanged := c.checkAndHotReloadConfig(c.nonHotReloadableConfig)
+	if isChanged && getEnvAsBool("RESTART_ON_CONFIG_CHANGE", false) {
+		os.Exit(1)
```

Comment:
@chandumlg @cisse21 do we really still need this functionality?

Reply:
We can get rid of it... we never actually used this and should be easy to add again if needed